server/src/test/scala/services/crunch/desklimits/flexed/FlexedTerminalDeskLimitsFromAvailableStaffSpec.scala

```diff
@@ -0,0 +1,209 @@
+        val existingNonEeaAllocation: Map[Queue, List[Int]] = Map(NonEeaDesk -> List.fill(24)(2))
+        val limits = FlexedTerminalDeskLimitsFromAvailableStaff(availableStaff, terminalDesks, Set(EeaDesk, NonEeaDesk), minDesksForEeaAndNonEea, Map())
+        val result = limits.maxDesksForMinutes(bstMidnightToMidnightByHour, EeaDesk, existingNonEeaAllocation)
+        val expected = List.fill(24)(3)
+
+        result === expected
+      }
+    }
+  }
+
+  "Given a staffing flexed desk limits provider with 1 fixed queue with min 1 desk, two flexed queues, each with 1 min desk, 10 flexed desks, 5 available staff, an existing Non-EEA allocation of 2 desks " >> {
+    "When I ask for max desks at each hour from midnight to midnight outside BST " >> {
+      "Then I should get 2 for every hour - the available staff minus the 2 allocations & 1 remaining min desk as it's lower than the max desks" >> {
+        val terminalDesks = List.fill(24)(10)
+        val availableStaff = List.fill(24)(5)
+        val minDesksForEeaAndNonEea: Map[Queue, IndexedSeq[Int]] = Map(EeaDesk -> minDesks, NonEeaDesk -> minDesks, EGate -> minDesks)
+        val existingNonEeaAllocation: Map[Queue, List[Int]] = Map(NonEeaDesk -> List.fill(24)(2))
+        val limits = FlexedTerminalDeskLimitsFromAvailableStaff(availableStaff, terminalDesks, Set(EeaDesk, NonEeaDesk), minDesksForEeaAndNonEea, Map())
+        val result = limits.maxDesksForMinutes(bstMidnightToMidnightByHour, EeaDesk, existingNonEeaAllocation)
+        val expected = List.fill(24)(2)
+
+        result === expected
+      }
+    }
+  }
+
+  "Given a staffing flexed desk limits provider with 1 fixed queue with min 1 desk, two flexed queues, each with 1 min desk, 10 flexed desks, 15 available staff, an existing Non-EEA allocation of 2 desks " >> {
+    "When I ask for max desks at each hour from midnight to midnight outside BST " >> {
+      "Then I should get 8 for every hour - the flexed desks minus the 2 flexed allocations regardless of the remaining non-flexed min desks as it's lower than the available staff" >> {
+        val terminalDesks = List.fill(24)(10)
+        val availableStaff = List.fill(24)(15)
+        val minDesksForEeaAndNonEea: Map[Queue, IndexedSeq[Int]] = Map(EeaDesk -> minDesks, NonEeaDesk -> minDesks, EGate -> minDesks)
+        val existingNonEeaAllocation: Map[Queue, List[Int]] = Map(NonEeaDesk -> List.fill(24)(2))
+        val limits = FlexedTerminalDeskLimitsFromAvailableStaff(availableStaff, terminalDesks, Set(EeaDesk, NonEeaDesk), minDesksForEeaAndNonEea, Map())
+        val result = limits.maxDesksForMinutes(bstMidnightToMidnightByHour, EeaDesk, existingNonEeaAllocation)
+        val expected = List.fill(24)(8)
+
+        result === expected
+      }
+    }
+  }
+
+//  "Given a staffing flexed desk limits provider with one flexed queue with flexed desks matching the hour (0 to 23) " >> {
```

Comment:
commented out tests here.

Reply:
Sorted. They were copied from another set of tests but I didn't end up needing them